url04f7fcf4e1_peteratt_HDFS_Replication_Simulator_tgz-pJ8-hdfs_replicationsimulator_MainJ8/src/hdfs/replicationsimulator/SimFileGenerator.java

```diff
@@ -0,0 +1,200 @@
+package hdfs.replicationsimulator;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class SimFileGenerator {
+
+	// Types:
+	// 0 is for constant
+	// 1 is for exponential
+	// 2 is for gaussian
+	private static final int CONSTANT = 0;
+	private static final int EXPONENTIAL = 1;
+	private static final int GAUSSIAN = 2;
+
+	private static String filename;
+
+	// Defaults
+	private static int distrType = 0;
+	private static int mttf = 30000;
+
+	private static int time;
+	private static int nodes;
+	private static int failingNodes;
+	private static int nBlocks;
+
+	private static List<Integer> failedNodes;
+
+	public static void main(String[] args) {
+		try {
+			parseArgs(args);
+		} catch (Exception e) {
+			System.err.println("Error while parsing parameters");
+			System.exit(1);
+		}
+	}
+
+	private static void parseArgs(String[] args) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 84-90 into a separate method.